src/device/index.ts

```diff
@@ -0,0 +1,102 @@
+import adb from 'adbkit'
+import delay from 'delay'
+import fs from 'fs'
+
+import poll from '../poll'
+import logger from '../logger'
+
+const client = adb.createClient()
+
+const REBOOT_TIMEOUT = 1000 * 60 * 2 // 2 minutes
+
+class Device {
+  public id: string
+
+  static async firstDeviceAvailable () {
+    const devices = await client.listDevices()
+    if (devices.length < 1) {
+      throw new Error('Could not find any devices :(')
+    }
+    const deviceId = devices[0].id
+    return new Device(deviceId)
+  }
+
+  constructor (deviceId: string) {
+    this.id = deviceId
```

Comment:
This confuses me a bit, so the only reason this is a class is to hold the device id? Could we not just inject the ID into most of these functions then we could test them a lot easier?

Reply:
Yep, the use of a class is just a style choice.

```typescript
const device = await Device.firstDeviceAvailable()
await.device.pullFile('/sdcard/window_dump.xml', destPath)
```

vs

```typescript
const deviceId = await device.firstDeviceIdAvailable()
await device.pullFile(deviceId, '/sdcard/window_dump.xml', destPath)
```

Reply:
I'd prefer to move these out of a class for maintainability sake. It's only one field and if we don't do it now we probably never will

Reply:
Ok, I'll do it tomorrow